controller/GiUnplannedItem.controller.js

```diff
@@ -0,0 +1,803 @@
+						return false;
+					}
+				}
+			}
+			return true;
+		},
+		clear: function() {
+			// Clear out appropriate fields
+			var oModel = this.getView().getModel();
+			oModel.setProperty("/Matnr", "");
+			oModel.setProperty("/Maktg", "");
+			oModel.setProperty("/Menge", "");
+			zController.liveChangeQty(null, 0);
+			oModel.setProperty("/Meins", "");
+			oModel.setProperty("/Lgpla", "");
+			oModel.setProperty("/Sgtxt", "");
+			oModel.setProperty("/Verme", "");
+			oModel.setProperty("/serialNumbers", []);
+			this.getView().getModel().setProperty("/serialNumbersRequired", false);
+		},
+		createSerialNumbersTable: function() {
+			// Create the item table via JS
+			snTable = new sap.m.Table({
+				//headerText: "Enter Serial Numbers",
+				width: "100%",
+				growing: true,
+				growingThreshold: 50,
+				noDataText: "Enter a Qty",
+				type: "Active" //enables itemPress event
+			});
+
+			//columns!!!
+			var inputCol = new sap.m.Column({
+				header: new sap.m.Label({
+					text: "Selected Serial Numbers"
+				}),
+				width: "75%"
+			});
+			var buttonCol = new sap.m.Column({
+				header: new sap.m.Label({
+					text: "Scan"
+				})
+			});
+			// rows!!!
+			var columnItems = new sap.m.ColumnListItem(this.getView().createId("snColumnListItem"), {
+				type: "Active" //enables itemPress event
+			});
+			//create cells to add to the rows.
+			var input = new sap.m.Input({
+				value: "{Sernr}",
+				placeholder: "S/N",
+				maxLength: 18,
+				editable: false
+			});
+			var button = new sap.m.Button({
+				width: "auto",
+				icon: "sap-icon://delete",
+				press: function(oEvent) {
+					var path = oEvent.getSource().getParent().getBindingContext().getPath();
+					var index = path.replace("/", "");
+					var sn = zController.getView().getModel().getProperty("/serialNumbers");
+					sn.splice(index, 1);
+					zController.getView().getModel().setProperty("/Menge", sn.length.toString());
+					snTable.getBinding("items").refresh();
+				}
+			});
+
+			// Add the columns to table
+			snTable.addColumn(inputCol);
+			snTable.addColumn(buttonCol);
+
+			columnItems.addCell(input);
+			columnItems.addCell(button);
+
+			this.getView().byId("vboxSN").addItem(snTable);
+
+			// Add a button for adding more serial numbers
+			var addButton = new sap.m.Button({
+				icon: "sap-icon://add",
+				press: zController.showSerialNumberSelection
+			});
+			this.getView().byId("vboxSN").addItem(addButton);
+		},
+		matnrScan: function(oEvent) {
+			if (!this.getView().getModel().getProperty("/Lgpla")) {
+				GlobalFuncs.showMessage("Error", "Enter a bin number", "ERROR");
+				return;
+			}
+			GlobalFuncs.scan(oEvent, zController.checkMaterial);
+		},
+		checkMaterial: function() {
+			var oModel = zController.getView().getModel();
+			var url = "/sap/opu/odata/sap/ZWM_GI_UNPLANNED_SRV/BinMaterialsSet?$filter=Lgnum eq '" + oModel.getProperty("/Lgnum") + "'" +

```

Comment:
besides the fact that we are using AJAX.... it is a good idea to put the logic for the backend calls in the models file, that way those can be reused and also the code of our controller will be cleaner